AmplifyPlugins/Storage/Sources/AWSS3StoragePlugin/Request/StorageDownloadDataRequest+Validate.swift

```diff
@@ -11,6 +11,7 @@ import Amplify
 extension StorageDownloadDataRequest {
     /// Performs client side validation and returns a `StorageError` for any validation failures.
     func validate() -> StorageError? {
+        guard path == nil else { return nil }
```

Comment:
I'm deferring the validation of when the operation is executed and the path is resolved since the validation is on the StoragePath extension itself and is an async call.  What do you think?

Reply:
The validation also depends on AuthService, so it made sense to me to pass use the auth service from the operation as well.

Reply:
I've added code comments on why `nil` is returned.